src/metabase/driver/common/parameters/values.clj

```diff
@@ -116,6 +121,17 @@
                  (sequential? value-info-or-infos) (mapv #(dissoc % :target) value-info-or-infos))
                i/no-value)})))
 
+(s/defn ^:private card-query-for-tag :- (s/maybe (s/cond-pre su/Map (s/eq i/no-value)))
+  "Returns the native query for the `:card` referenced by the given tag."
+  [tag :- TagParam, _params :- (s/maybe [i/ParamValue])]
+  (when-let [card-id (:card tag)]
+    (when-let [query (db/select-one-field :dataset_query Card :id card-id)]
+      (i/map->CardQuery
+       {:card-id card-id
+        :query   (condp = (name (:type query))
```

Comment:
why going through `name` rather than comparing keywords?

Reply:
Is [this](https://github.com/metabase/metabase/pull/11835/files#diff-5b28fb9bc94ad945da73a4fad99fc434R131) what you had in mind?

Reply:
Yes